packages/api/amplify_api/lib/src/method_channel_api.dart

```diff
@@ -284,55 +316,119 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
+        apiName: apiName);
   }
 
   @override
-  RestOperation put({required RestOptions restOptions}) {
-    return _restFunctionHelper(methodName: 'put', restOptions: restOptions);
+  CancelableOperation<AWSStreamedHttpResponse> put(
+    String path, {
+    HttpPayload? body,
+    Map<String, String>? headers,
+    Map<String, String>? queryParameters,
+    String? apiName,
+  }) {
+    return _restFunctionHelper(
+        methodName: 'put',
+        path: path,
+        body: body,
+        headers: headers,
+        queryParameters: queryParameters,
+        apiName: apiName);
   }
 
   @override
-  RestOperation post({required RestOptions restOptions}) {
-    return _restFunctionHelper(methodName: 'post', restOptions: restOptions);
+  CancelableOperation<AWSStreamedHttpResponse> post(
+    String path, {
+    HttpPayload? body,
+    Map<String, String>? headers,
+    Map<String, String>? queryParameters,
+    String? apiName,
+  }) {
+    return _restFunctionHelper(
+        methodName: 'post',
+        path: path,
+        body: body,
+        headers: headers,
+        queryParameters: queryParameters,
+        apiName: apiName);
   }
 
   @override
-  RestOperation delete({required RestOptions restOptions}) {
-    return _restFunctionHelper(methodName: 'delete', restOptions: restOptions);
+  CancelableOperation<AWSStreamedHttpResponse> delete(
+    String path, {
+    HttpPayload? body,
+    Map<String, String>? headers,
+    Map<String, String>? queryParameters,
+    String? apiName,
+  }) {
+    return _restFunctionHelper(
+        methodName: 'delete',
+        path: path,
+        body: body,
+        headers: headers,
+        queryParameters: queryParameters,
+        apiName: apiName);
   }
 
   @override
-  RestOperation head({required RestOptions restOptions}) {
-    return _restFunctionHelper(methodName: 'head', restOptions: restOptions);
+  CancelableOperation<AWSStreamedHttpResponse> head(
+    String path, {
+    Map<String, String>? headers,
+    Map<String, String>? queryParameters,
+    String? apiName,
+  }) {
+    return _restFunctionHelper(
+        methodName: 'head',
+        path: path,
+        headers: headers,
+        queryParameters: queryParameters,
+        apiName: apiName);
   }
 
   @override
-  RestOperation patch({required RestOptions restOptions}) {
-    return _restFunctionHelper(methodName: 'patch', restOptions: restOptions);
+  CancelableOperation<AWSStreamedHttpResponse> patch(
+    String path, {
+    HttpPayload? body,
+    Map<String, String>? headers,
+    Map<String, String>? queryParameters,
+    String? apiName,
+  }) {
+    return _restFunctionHelper(
+        methodName: 'patch',
+        path: path,
+        body: body,
+        headers: headers,
+        queryParameters: queryParameters,
+        apiName: apiName);
   }
 
-  @override
+  /// Cancels a request with a given request ID.
+  @Deprecated('Use .cancel() on CancelableOperation instead.')
```

Comment:
Maybe another candidate for auto migration? See other comments about dart fix.

Reply:
Based on my understanding of auto-migration tools (still limited) I think this one is harder to auto-migrate because it's not a simple transform, but rather, would require people to change logic in how they're managing cancelations. Instead of managing IDs somewhere and passing them, would have to manage the operations.